inference-engine/src/transformations/include/transformations/smart_reshape/softmax_sr.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <functional>
+
+#include <transformations_visibility.hpp>
+
+#include <ngraph/pass/graph_rewrite.hpp>
+
+namespace ngraph {
+namespace pass {
+
+class TRANSFORMATIONS_API ReshapeSoftMaxReshape;
+
+}  // namespace pass
+}  // namespace ngraph
+
+class ngraph::pass::ReshapeSoftMaxReshape: public ngraph::pass::MatcherPass {
```

Comment:
the idea was to hide all these "sub-transformations" which are part of umbrella "SmartReshape" from an end user / transformations API. I think only `SmartReshape` is supposed to be an API, is not it?

And in tests we have to use `SmartReshape` instead of `ReshapeSoftMaxReshape` in this case, because last one would not be exported.

Reply:
The thing is we will release them separately in 2021.3, it is only for now they are presented as solid set of transformations, but later user will have an option to pick and apply them separately. Later on, we will have smart reshape transformations that would not be executed by default. So what is our plan then? Hide them till 2021.3? Or allow user to play with them already?

Reply:
Regarding the testing -- transformations may interfere, so I would like to have both unit-test like test scenario with only one transformation involved and integration test with SmartReshape transformation call or even CNNNetwork creation as we already have for SmartReshape transformations for MatMul.

Since there is not that much transformations, we have no intersections of their region of interest. Ans that is why we don't have that much "integration" tests.

Reply:
Exporting the API only for testing is not a good argument.. 

I just think if we export all the transformations - this will be never be refactored. But up to you..